shared/actions/config/index.js

```diff
@@ -206,3 +207,99 @@ function getCurrentStatus (): AsyncAction {
     })
   }
 }
+
+type Info = {
+  url: ?string,
+  callbacks: Array<(url: ?string) => void>,
+  requested: boolean,
+  done: boolean,
+  error: boolean,
+}
+
+const _usernameToURL: {[key: string]: ?Info} = {}
+
+const _getUserImages = throttle(() => {
+  const usersToResolve = Object.keys(_usernameToURL).filter(username => {
```

Comment:
This iteration will get slower as more avatars are loaded. Worth breaking out pending into a separate data structure?